gapii/cc/spy_base.h

```diff
@@ -89,6 +89,10 @@ class SpyBase {
         return should_trace(api) ? mEncoder : mNullEncoder;
     }
 
+    std::shared_ptr<gapii::PackEncoder> parentEncoder(uint8_t api, CallObserver* parent) {
```

Comment:
This feels like a bit of a clunky method. There's nothing particularly parent-specific about the implementation, and the name has no real indication of what it does. 
I wonder if it would be clearer to just expose `mNullEncoder` and let the caller do this logic. This would also deobfuscate the call to `should_trace`.